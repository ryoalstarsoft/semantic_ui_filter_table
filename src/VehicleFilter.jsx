import React from 'react';
import PropTypes from 'prop-types';
import {Form, Popup} from 'semantic-ui-react';

const regex = new RegExp("^[a-zA-Z0-9 ]+$");

export class VehicleFilter extends React.Component {
  constructor() {
    super();
    this.state = {
      filter: '',
      filterValid: true,
    };
    this.handleOnChange = this.handleOnChange.bind(this);
  }

  handleOnChange(event, {name, value}) {
    if (value !== '' && !regex.test(value)) {
      this.setState({[name]: value, filterValid: false});
    } else {
      this.setState({[name]: value, filterValid: true});
      this.props.onSubmitFilter(value);
    }
  }

  render() {
    const { filter } = this.state;
    let popupMessage = '';
    if (!this.state.filterValid) {
      popupMessage = 'Invalid character.';
    } else if (this.props.totalCount === 0) {
      popupMessage = 'No results found.'
    }

    return (
      <Form>
        <Form.Group>
          <Form.Field>
            <Popup
              trigger={<Form.Input
                placeholder='Enter the filter.'
                name='filter'
                value={filter}
                error={!this.state.filterValid}
                label='Filter'
                onChange={this.handleOnChange}
                icon='search'
                loading={this.props.loading}
              />}
              content={popupMessage}
              on='click'
              open={!this.state.filterValid || this.props.totalCount === 0}
              position='right center'
            />
          </Form.Field>
        </Form.Group>
      </Form>
    )
  }
}

VehicleFilter.propTypes = {
  onSubmitFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
  totalCount: PropTypes.number.isRequired,
};
