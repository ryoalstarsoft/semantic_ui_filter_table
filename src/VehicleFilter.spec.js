import React from 'react';
import {shallow} from 'enzyme';
import {VehicleFilter} from './VehicleFilter';

describe('VehicleFilter', () => {
  it('should render correctly in "debug" mode', () => {
    let filter = '';
    let totalCount = 100;
    let onSubmitFilter = function() {};

    const component = shallow(<VehicleFilter
      filter = { filter }
      totalCount = { totalCount }
      onSubmitFilter = { onSubmitFilter }
      debug />);

    expect(component).toMatchSnapshot();

    component.instance().handleOnChange({}, {});
  });
});
