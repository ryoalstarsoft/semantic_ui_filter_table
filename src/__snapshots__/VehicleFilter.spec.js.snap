// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleFilter should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehicleFilter
    debug={true}
    filter=""
    onSubmitFilter={[Function]}
    totalCount={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": "form",
      "children": <FormGroup>
        <FormField>
          <Popup
            content=""
            keepInViewPort={true}
            on="click"
            open={false}
            position="right center"
            trigger={
              <FormInput
                as={[Function]}
                control={[Function]}
                error={false}
                icon="search"
                label="Filter"
                name="filter"
                onChange={[Function]}
                placeholder="Enter the filter."
                value=""
              />
            }
          />
        </FormField>
      </FormGroup>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <FormField>
          <Popup
            content=""
            keepInViewPort={true}
            on="click"
            open={false}
            position="right center"
            trigger={
              <FormInput
                as={[Function]}
                control={[Function]}
                error={false}
                icon="search"
                label="Filter"
                name="filter"
                onChange={[Function]}
                placeholder="Enter the filter."
                value=""
              />
            }
          />
        </FormField>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Popup
            content=""
            keepInViewPort={true}
            on="click"
            open={false}
            position="right center"
            trigger={
              <FormInput
                as={[Function]}
                control={[Function]}
                error={false}
                icon="search"
                label="Filter"
                name="filter"
                onChange={[Function]}
                placeholder="Enter the filter."
                value=""
              />
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "content": "",
            "keepInViewPort": true,
            "on": "click",
            "open": false,
            "position": "right center",
            "trigger": <FormInput
              as={[Function]}
              control={[Function]}
              error={false}
              icon="search"
              label="Filter"
              name="filter"
              onChange={[Function]}
              placeholder="Enter the filter."
              value=""
            />,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "form",
        "children": <FormGroup>
          <FormField>
            <Popup
              content=""
              keepInViewPort={true}
              on="click"
              open={false}
              position="right center"
              trigger={
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  error={false}
                  icon="search"
                  label="Filter"
                  name="filter"
                  onChange={[Function]}
                  placeholder="Enter the filter."
                  value=""
                />
              }
            />
          </FormField>
        </FormGroup>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FormField>
            <Popup
              content=""
              keepInViewPort={true}
              on="click"
              open={false}
              position="right center"
              trigger={
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  error={false}
                  icon="search"
                  label="Filter"
                  name="filter"
                  onChange={[Function]}
                  placeholder="Enter the filter."
                  value=""
                />
              }
            />
          </FormField>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Popup
              content=""
              keepInViewPort={true}
              on="click"
              open={false}
              position="right center"
              trigger={
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  error={false}
                  icon="search"
                  label="Filter"
                  name="filter"
                  onChange={[Function]}
                  placeholder="Enter the filter."
                  value=""
                />
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content": "",
              "keepInViewPort": true,
              "on": "click",
              "open": false,
              "position": "right center",
              "trigger": <FormInput
                as={[Function]}
                control={[Function]}
                error={false}
                icon="search"
                label="Filter"
                name="filter"
                onChange={[Function]}
                placeholder="Enter the filter."
                value=""
              />,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
